src/scripts/quests/questTypes/GainShardsQuest.ts

```diff
@@ -10,25 +10,37 @@ class GainShardsQuest extends Quest implements QuestInterface {
         this.focus = App.game.statistics.shardsGained[this.type];
     }
 
+    public static rewardWeight: Record<PokemonType, number> = {
+        [PokemonType.None]:     0,
+        [PokemonType.Normal]:   1,
+        [PokemonType.Fire]:     2,
+        [PokemonType.Water]:    1,
+        [PokemonType.Electric]: 2,
+        [PokemonType.Grass]:    2,
+        [PokemonType.Ice]:      4,
+        [PokemonType.Fighting]: 1,
+        [PokemonType.Poison]:   1,
+        [PokemonType.Ground]:   2,
+        [PokemonType.Flying]:   1,
+        [PokemonType.Psychic]:  4,
+        [PokemonType.Bug]:      3,
+        [PokemonType.Rock]:     2,
+        [PokemonType.Ghost]:    4,
+        [PokemonType.Dragon]:   5,
+        [PokemonType.Dark]:     3,
+        [PokemonType.Steel]:    3,
+        [PokemonType.Fairy]:    5,
+    }
+
     public static generateData(): any[] {
-        const possibleTypes = [
-            PokemonType.Normal,
-            PokemonType.Poison,
-            PokemonType.Water,
-            PokemonType.Grass,
-            PokemonType.Flying,
-            PokemonType.Fire,
-            PokemonType.Fighting,
-        ];
-        const type = SeededRand.fromArray(possibleTypes);
+        const type = SeededRand.fromArray(GameHelper.createArray(1, GameHelper.enumLength(PokemonType) - 1, 1));
```

Comment:
This looks like it will be able to generate the type `18` which is `undefined`?
```js
GameHelper.createArray(1, GameHelper.enumLength(PokemonType) - 1, 1)
(18) [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]
```

Reply:
Fixed to
```JS
const type = SeededRand.fromArray(GameHelper.createArray(0, GameHelper.enumLength(PokemonType) - 2, 1));
```

If you find a cleaner solution feel free to add it. I don't know of a clean way of not including the `None` enum value.

Reply:
Just realized we already have a method in `SeededRand` actually:
```
SeededRand.fromEnum(PokemonType);
```
Which filters out any strings and numbers below 0.